### ISSUE TITLE
New Account DelayDays Validation Error

### ISSUE THREAD

**Issue by jheth** [stripe/stripe-mock#133]
When using PayoutSchedule DelayDaysMinimum parameter I get the following error:

```go
params.PayoutSchedule.Interval = stripe.String("daily")
params.PayoutSchedule.DelayDays = stripe.Int64(0)
params.PayoutSchedule.DelayDaysMinimum = stripe.Bool(true)
```

Request: POST /v1/accounts
Request data = map[external_account:tok_1DoUTXYZ legal_entity:map[type:individual address:map[state:PA city:Pandatown country:US line1:123 Panda Lane line2:Apt 5 postal_code:76543] dob:map[day:1 month:1 year:1980] first_name:John last_name:Doe personal_id_number:000000000 ssn_last_4:0000] payout_schedule:map[delay_days:minimum interval:daily] support_email:yatn4kqptl44qk2vrao7p5skpmskzxvp@testing.com tos_acceptance:map[date:1547240600 ip:127.0.0.1 user_agent:Go-http-client/1.1] expand:[legal_entity.verification] debit_negative_balances:true email:yatn4kqptl44qk2vrao7p5skpmskzxvp@testing.com type:custom country:US support_phone:800-123-1234]

Request validation error: validator 0xc000999650 failed: object property 'payout_schedule' validation failed: object property 'delay_days' validation failed: value is not numeric
Response: elapsed=392.876µs status=400

**Comment by jheth** [comment]
It works if I don't set DelayDaysMinimum, sending `payout_schedule:map[delay_days:0 interval:daily]` to stripe-mock.  However, that causes the real/test stripe API to fail with the error:

{
 “status”:400, 
  “message”: “You cannot lower this merchant’s delay below 2”,
  “param”: “payout_schedule[delay_days]“,
  ”request_id”: “req_X”,
  “type”: “invalid_request_error”
}

**Comment by brandur-stripe** [comment]
Hi @jheth, would you mind trying to upgrade your version of stripe-mock and trying again? It's still changing a lot, and I think the issue you were running into is solved now.

I'm able to successfully simulate an account create against stripe-mock with this Ruby script, which I believe is sending similar parameters to your integration:

``` ruby
require 'stripe'

Stripe.api_key = "sk_test_123"
Stripe.api_base = "http://localhost:12111"

account = Stripe::Account.create(
  payout_schedule: {
    delay_days: 0,
    delay_days_minimum: true,
    interval: "daily",
  },
  type: "standard"
)
p account
```

Encoded parameters:

```
INFO Request to Stripe API method=post num_retries=0 path=/v1/accounts
DEBU Request details body="payout_schedule[delay_days]=0&payout_schedule[delay_days_minimum]=true&payout_schedule[interval]=daily&type=standard"
```

**Comment by jheth** [comment]
This was running against the latest stripe-mock.  I would expect the request body to be what you show, where delay_days=0 and delay_days_minimum=true.

When I print the map in go it shows, `delay_days:minimum` which would not be a valid integer.

payout_schedule:map[delay_days:minimum interval:daily]

**Comment by ob-stripe** [comment]
@jheth It sounds like you're using stripe-go to send the request, is that correct? Can you share the exact code you're using to create the parameters and send the request, and also tell us the versions of Go and stripe-go you're using? (I wonder if this is a client issue rather than a stripe-mock one.)

**Comment by jheth** [comment]
Yes, absolutely. I included the 3 lines of PayoutSchedule configuration in the description.

This is the function in stripe-go that is overwriting delay_days to be `minimum` instead of 0.

```
func (p *PayoutScheduleParams) AppendTo(body *form.Values, keyParts []string) {
	if BoolValue(p.DelayDaysMinimum) {
		body.Add(form.FormatKey(append(keyParts, "delay_days")), "minimum")
	}
}
```

**Comment by jheth** [comment]
Oh, I may see what you're saying, I am NOT using the latest stripe-go package. That is likely it. I'll check for an update there.

**Comment by jheth** [comment]
The latest stripe-go package has the same function overwriting `delay_days`, which I think is the issue.

**Comment by ob-stripe** [comment]
@jheth Thanks for checking. Would you mind opening an issue on the stripe-go repository, and referencing this one?

**Comment by brandur-stripe** [comment]
Doh, actually on a second look it looks like [from the documentation](https://stripe.com/docs/api/accounts/create?&lang=go#create_account-payout_schedule-delay_days) what stripe-go is doing here is correct, and this is an OpenAPI/stripe-mock problem.

I'll check to see whether it's relatively tractable to get this fixed. Sorry about the run around!

**Comment by brandur-stripe** [comment]
Okay, just waiting on a review now, but I have a fix coming down the pipe. It should be out in the next day or so.

**Comment by ob-stripe** [comment]
Oops, thanks @brandur-stripe and sorry for misleading you @jheth!

**Comment by brandur-stripe** [comment]
Hey Joe, I just released 0.41.0 and confirmed that a call that was previously failing now succeeds. Please upgrade and try it out!

**Comment by jheth** [comment]
Thanks @brandur-stripe! It's all good now.
